lib/mix/lib/mix/tasks/archive.build.ex

```diff
@@ -42,79 +42,103 @@ defmodule Mix.Tasks.Archive.Build do
     * `--include-dot-files` - adds dot files from priv directory to the archive.
 
   """
-  @switches [force: :boolean, compile: :boolean, output: :string, input: :string,
-             deps_check: :boolean, archives_check: :boolean, elixir_version_check: :boolean,
-             include_dot_files: :boolean]
+  @switches [
+    force: :boolean,
+    compile: :boolean,
+    output: :string,
+    input: :string,
+    deps_check: :boolean,
+    archives_check: :boolean,
+    elixir_version_check: :boolean,
+    include_dot_files: :boolean
+  ]
 
   def run(args) do
     {opts, _} = OptionParser.parse!(args, aliases: [o: :output, i: :input], strict: @switches)
 
-    project = Mix.Project.get
+    project = Mix.Project.get()
 
     if project && Keyword.get(opts, :compile, true) do
-      Mix.Task.run :compile, args
+      Mix.Task.run(:compile, args)
     end
 
-    source = cond do
-      input = opts[:input] ->
-        input
-      project ->
-        path = Mix.Project.app_path
-        if elixir = Mix.Project.config[:elixir] do
-          File.write Path.join(path, ".elixir"), elixir
-        else
-          File.rm Path.join(path, ".elixir")
-        end
-        path
-      true ->
-        Mix.raise "Cannot create archive without input directory, " <>
-          "please pass -i as an option"
-    end
+    source =
+      cond do
+        input = opts[:input] ->
+          input
 
-    project_config = Mix.Project.config
-    target = cond do
-      output = opts[:output] ->
-        output
-      project_config[:app] ->
-        Mix.Local.name_for(:archive, project_config)
-      true ->
-        Mix.raise "Cannot create archive without output file, " <>
-          "please pass -o as an option"
-    end
+        project ->
+          path = Mix.Project.app_path()
+
+          if elixir = Mix.Project.config()[:elixir] do
+            File.write(Path.join(path, ".elixir"), elixir)
+          else
+            File.rm(Path.join(path, ".elixir"))
+          end
+
+          path
+
+        true ->
+          Mix.raise(
+            "Cannot create archive without input directory, " <> "please pass -i as an option"
```

Comment:
Please remove concatenation. If it still doesn't git on a single line, please extract to a variable.